ee/tabby-webserver/src/service/background_job/web_crawler.rs

```diff
@@ -46,4 +48,21 @@ impl WebCrawlerJob {
         indexer.commit();
         Ok(())
     }
+    pub async fn run(self, embedding: Arc<dyn Embedding>) -> tabby_schema::Result<()> {
+        let url = self.url.clone();
+        if tokio::time::timeout(
+            Duration::from_secs(CRAWLER_TIMEOUT_SECS),
+            self.run_impl(embedding),
+        )
+        .await
+        .is_err()
```

Comment:
can this method being somehow tested? can you add a unit test to demonstrate the behavior? e.g a async function with async for loop, and can be halted by timeout.